### PR TITLE
Add FileBasedTableGenerator::_add_constraints & fix metadata in testdata/imdb_sample

### PR DESCRIPTION
**Pull request by julianmenzler**
This PR adds a callback in `FileBasedTableGenerator` to allow for constraints in, e.g., the JoinOrderBenchmark.

Also, it fixes wrong JOB metadata (column nullability) in `resources/test_data/imdb_sample/`.

### CODE REVIEW

File: src/benchmarklib/file_based_table_generator.hpp [hunk 1]
```diff
@@ -13,8 +14,17 @@ class FileBasedTableGenerator : virtual public AbstractTableGenerator {
 
   std::unordered_map<std::string, BenchmarkTableInfo> generate() override;
 
+  /**
+   * @param add_constraints_callback is called after the table data has been added to the StorageManager.
```
Comment (Bensk1) [inline src/benchmarklib/file_based_table_generator.hpp:18]: Maybe add "... is called (via `_add_constraints()`) after ..."

**Review by Bensk1 — COMMENTED** [review]
Looks good to me. Since we do not run JCCH in the CI, have you tried it manually? And we should fix the changes in setup_imdb.py that do not belong to this PR.

File: src/benchmarklib/file_based_table_generator.hpp (hunk 1 above)
Comment (julianmenzler) [reply src/benchmarklib/file_based_table_generator.hpp:18]:
```cpp
  /**
   * Set @param add_constraints_callback to define table constraints, if available. It is called by _add_constraints
   * after tables have been generated.
   */
```

Better?

**Comment by julianmenzler** [comment]
> And we should fix the changes in setup_imdb.py that do not belong to this PR.

Fixed it by merging origin/master.

> Since we do not run JCCH in the CI, have you tried it manually?

Good hint. I thought this was already part of one CI stage. I start a run and will report back.

**Comment by julianmenzler** [comment]
Looks fine. 

<details>
<summary>Configuration Overview - click to expand</summary>

```diff
 +Configuration Overview----+------------------------------------------------+------------------------------------------------+
 | Parameter                | jcch_master.json                               | jcch_2368.json                                 |
 +--------------------------+------------------------------------------------+------------------------------------------------+
 |  GIT-HASH                | 6b3fcde6d6e663fa06a62c4d051bd7495bb2a33a-dirty | 3175009f8947a80b71a67d3c41bb5aa276ef5b99-dirty |
 |  benchmark_mode          | Ordered                                        | Ordered                                        |
 |  build_type              | release                                        | release                                        |
 |  chunk_size              | 65535                                          | 65535                                          |
 |  clients                 | 1                                              | 1                                              |
 |  clustering              | None                                           | None                                           |
 |  compiler                | gcc 9.2                                        | gcc 9.2                                        |
 |  cores                   | 0                                              | 0                                              |
 |  date                    | 2021-06-03 13:04:54                            | 2021-06-03 12:24:38                            |
 |  encoding                | {'default': {'encoding': 'Dictionary'}}        | {'default': {'encoding': 'Dictionary'}}        |
 |  indexes                 | False                                          | False                                          |
 |  max_duration            | 60000000000                                    | 60000000000                                    |
 |  max_runs                | -1                                             | -1                                             |
 |  scale_factor            | 1.0                                            | 1.0                                            |
 |  time_unit               | ns                                             | ns                                             |
 |  use_prepared_statements | False                                          | False                                          |
 |  using_scheduler         | False                                          | False                                          |
 |  verify                  | False                                          | False                                          |
 |  warmup_duration         | 0                                              | 0                                              |
 +--------------------------+------------------------------------------------+------------------------------------------------+
```
</details>

```diff
 +-------------------++----------+---------+--------++----------+----------+--------+---------+
 | Item              || Latency (ms/iter)  | Change || Throughput (iter/s) | Change | p-value |
 |                   ||      old |     new |        ||      old |      new |        |         |
 +-------------------++----------+---------+--------++----------+----------+--------+---------+
 | JCC-H (skewed) 01 ||   1026.7 |  1004.9 |   -2%  ||     0.97 |     1.00 |   +2%  |  0.1140 |
 | JCC-H (skewed) 02 ||     18.2 |    18.6 |   +2%  ||    54.99 |    53.81 |   -2%  |  0.0000 |
 | JCC-H (skewed) 03 ||    179.4 |   182.6 |   +2%  ||     5.57 |     5.48 |   -2%  |  0.5452 |
 | JCC-H (skewed) 04 ||    209.1 |   205.7 |   -2%  ||     4.78 |     4.86 |   +2%  |  0.7033 |
 | JCC-H (skewed) 05 ||    354.6 |   366.2 |   +3%  ||     2.82 |     2.73 |   -3%  |  0.0000 |
 | JCC-H (skewed) 06 ||     27.2 |    27.6 |   +1%  ||    36.71 |    36.26 |   -1%  |  0.0000 |
 | JCC-H (skewed) 07 ||    654.4 |   667.2 |   +2%  ||     1.53 |     1.50 |   -2%  |  0.0848 |
+| JCC-H (skewed) 08 ||     15.8 |    14.7 |   -7%  ||    63.36 |    68.16 |   +8%  |  0.0000 |
 | JCC-H (skewed) 10 ||    278.6 |   276.8 |   -1%  ||     3.59 |     3.61 |   +1%  |  0.8862 |
 | JCC-H (skewed) 11 ||     27.6 |    28.3 |   +2%  ||    36.21 |    35.33 |   -2%  |  0.0000 |
 | JCC-H (skewed) 12 ||    175.6 |   179.4 |   +2%  ||     5.70 |     5.57 |   -2%  |  0.0000 |
 | JCC-H (skewed) 13 ||    582.4 |   592.8 |   +2%  ||     1.72 |     1.69 |   -2%  |  0.0001 |
 | JCC-H (skewed) 14 ||     91.4 |    91.5 |   +0%  ||    10.94 |    10.93 |   -0%  |  0.9661 |
 | JCC-H (skewed) 15 ||     51.0 |    51.4 |   +1%  ||    19.59 |    19.44 |   -1%  |  0.0548 |
 | JCC-H (skewed) 16 ||    194.0 |   197.2 |   +2%  ||     5.16 |     5.07 |   -2%  |  0.0001 |
 | JCC-H (skewed) 17 ||    535.5 |   539.1 |   +1%  ||     1.87 |     1.85 |   -1%  |  0.1370 |
 | JCC-H (skewed) 18 ||    661.4 |   662.5 |   +0%  ||     1.51 |     1.51 |   -0%  |  0.7146 |
 | JCC-H (skewed) 19 ||    148.0 |   148.2 |   +0%  ||     6.76 |     6.75 |   -0%  |  0.8298 |
 | JCC-H (skewed) 20 ||    426.9 |   431.1 |   +1%  ||     2.34 |     2.32 |   -1%  |  0.0318 |
 | JCC-H (skewed) 21 ||   1038.4 |  1032.9 |   -1%  ||     0.96 |     0.97 |   +1%  |  0.0713 |
 | JCC-H (skewed) 22 ||     60.7 |    61.6 |   +1%  ||    16.46 |    16.24 |   -1%  |  0.0002 |
 +-------------------++----------+---------+--------++----------+----------+--------+---------+
 | Sum               ||   6756.9 |  6780.3 |   +0%  ||          |          |        |         |
 | Geomean           ||          |         |        ||          |          |   -0%  |         |
 +-------------------++----------+---------+--------++----------+----------+--------+---------+
```

(JCCH Q9 gets killed for both branches, master and this PR's branch)

File: src/benchmarklib/file_based_table_generator.hpp (hunk 1 above)
Comment (Bensk1) [reply src/benchmarklib/file_based_table_generator.hpp]: Perfect.

**Review by Bensk1 — APPROVED** [review]

**Comment by Bouncner** [comment]
> (JCCH Q9 gets killed for both branches, master and this PR's branch)

Any idea what's happening there?

**Comment by julianmenzler** [comment]
```
- Benchmarking JCC-H (skewed) 09

Warning: JCC-H Query 9 needs optimization. Consider skipping it using -q

[PERF] Build side larger than probe side in hash join at src/lib/operators/join_hash.hpp:109
        Performance can be affected. This warning is only shown once.

[PERF] Unresolved iterator created for AbstractPosList at src/lib/storage/pos_lists/abstract_pos_list.cpp:5
        Performance can be affected. This warning is only shown once.

Killed
```

Timeout, I guess.

**Comment by Bensk1** [comment]
I believe one of the joins consumes an insane amount of memory due to large result sizes.

**Pull request merged**